### PR TITLE
docs: add installation vignette

### PR DESCRIPTION
**Pull request by eitsupi**
Part of #485

After merging this, I will remove installation descriptions from the README in the follow-up PR.

### CODE REVIEW

File: vignettes/install.Rmd [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  comment = "#>"
+)
+
+rust_toolchain_version = read.dcf(
+  system.file("DESCRIPTION", package = "polars"),
+  fields = "Config/polars/RustToolchainVersion", all = TRUE
+)[1, 1]
+```
+
+The `polars` package can be installed from R-universe, or GitHub.
```
Comment (etiennebacher) [inline vignettes/install.Rmd:22]:
```suggestion
The `polars` package can be installed from R-universe or GitHub.
```

File: vignettes/install.Rmd [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+Installing the latest release version.
+
+```r
+Sys.setenv(NOT_CRAN = "true") # Enable installation with pre-built Rust library binary, or enable Rust caching
+install.packages("polars", repos = "https://rpolars.r-universe.dev")
+```
+
+- On amd64 architecture Windows and macOS, binary R packages will be installed.
+- On the other platforms, the pre-built Rust library binary will be downloaded while building the R source package.
+- If the pre-built Rust library binary is not available, the Rust library will be built from source.
```
Comment (etiennebacher) [inline vignettes/install.Rmd:39]:
```suggestion
- If the pre-built Rust library binary is not available, the Rust library will be built from source (provided that Rust is installed).
```

**Review by etiennebacher — APPROVED** [review]
LGTM, a few minor comments. I think we should keep the "install from R-universe" also in the README, since it should cover most usages

**Comment by eitsupi** [comment]
@etiennebacher The install vignette does not seem rendered on the website. Any thoughts?

**Comment by etiennebacher** [comment]
I'll take a look, it also looks like the performance vignette was not updated since last week

**Pull request merged**